Update service.py

lj894 opened the pull request:
Bug fix to remove serialize() call.
Removed helper class and did code clean up to match the style of other functions.

lj894 committed:
Update service.py

Bug fix to remove serialize() call.
Removed helper class and did code clean up to match the style of other functions.

sushanthsamala commented:
@lj894 Since, you opened a new pull request for this task and my latest changes are merged. I would suggest you to add in unit tests for the end point that you added and the related model.py functions. Currently there is no check if "id" is int or not a negative number or not etc.

lj894 committed:
Merge branch 'master' into fix-service-error
Get latest code from master to add unit test functions

MinervaMint commented:
I think the `find` method in model.py may need to be modified. Now it uses `cls.query.get(by_id)` which did not work when I was running it locally - I think it is bacause that `id` is not the only primary key, and maybe `cls.query.filter` is what is needed?

lj894 committed:
Merge branch 'master' into fix-service-error

lj894 committed:
added test cases for query by id API

Added test cases for get related products by id function.

lj894 committed:
Merge branch 'master' into fix-service-error
merge latest master code to fix-service-error branch

lj894 committed:
update test case

Created more test data before running the test to increase code coverage

lj894 commented:
Updated the code and added the test cases. Please review again.

sushanthsamala commented on tests/test_service.py:
```diff
@@ -75,6 +75,33 @@ def test_index(self):
+            ("/recommendations/" + str(recommendation[0][0].id))
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(isinstance(resp, object), True, "Received incorrect recommendation")
```
Great work so far. Can you check if the returned resp.get_json() has recommendation objects that have id == recommendation[0][0].id and the number of objects is equal to the number of objects with product-id == recommendation[0][0].id created in line 81 test_service.py?

lj894 committed:
Added new tests

Added test for check newly created id exists in the result, and see if the total number of records match the number of created records.

sushanthsamala commented on tests/test_service.py:
```diff
@@ -75,6 +75,47 @@ def test_index(self):
+
+        num_records = len(resp.get_json()[0]["ids"]) + len(resp.get_json()[1]["ids"]) + len(resp.get_json()[2]["ids"])
+
+        self.assertEqual(num_records, 1, "Received incorrect records")
```
I'm wondering why the num_records should be equal to 1?

lj894 replied:
I saw in RecommendationFactory, a new id is generated every time the class initialized:  `id = FuzzyChoice(choices=range(1, 10001))`. So when the code calls `self._create_recommendations(count=20, by_status=True)`. it creates 20 records with different ids. Therefore when you use the API to query a given id, only 1 matching record can be found among one of the 3 relationship_types.

sushanthsamala replied:
Oh okay makes sense
